src/transformers/data/processors/utils.py

```diff
@@ -39,23 +43,14 @@ class InputExample(object):
             specified for train and dev examples, but not for test examples.
     """
 
-    def __init__(self, guid, text_a, text_b=None, label=None):
-        self.guid = guid
-        self.text_a = text_a
-        self.text_b = text_b
-        self.label = label
-
-    def __repr__(self):
-        return str(self.to_json_string())
-
-    def to_dict(self):
-        """Serializes this instance to a Python dictionary."""
-        output = copy.deepcopy(self.__dict__)
-        return output
+    guid: str
+    text_a: str
+    text_b: Optional[str] = None
+    label: Optional[str] = None
 
     def to_json_string(self):
         """Serializes this instance to a JSON string."""
-        return json.dumps(self.to_dict(), indent=2, sort_keys=True) + "\n"
+        return json.dumps(dataclasses.asdict(self), indent=2, sort_keys=True) + "\n"
```

Comment:
we do not do a deepcopy anymore because the data structure is immutable anyways.